### PR TITLE
services: forbid https -> http Oracle request auto-redirect

### PR DESCRIPTION
**Pull request by AnnaShaleva**

### CODE REVIEW

File: pkg/services/oracle/network.go [hunk 1]
```diff
@@ -89,6 +89,10 @@ func getDefaultClient(cfg config.OracleConfiguration) *http.Client {
 	// Override default CheckRedirect behaviour so that there's no restriction on the
 	// number of redirects (10 at max is allowed by default).
 	client.CheckRedirect = func(req *http.Request, via []*http.Request) error {
+		if len(via) > 0 && via[0].URL.Scheme == "https" && req.URL.Scheme != "https" {
```
Comment (AnnaShaleva) [inline pkg/services/oracle/network.go:92]: @roman-khimov, it worked like this in C# before https://github.com/neo-project/neo-modules/pull/692 merge (see https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclienthandler.allowautoredirect?view=net-6.0#remarks). But I'm not sure it behaves like this now.

File: pkg/services/oracle/network.go [hunk 2]
```diff
@@ -86,10 +86,9 @@ func getDefaultClient(cfg config.OracleConfiguration) *http.Client {
 		DialContext: d.DialContext,
 	}
 	client.Timeout = cfg.RequestTimeout
+	// Override default CheckRedirect behaviour so that there's no restriction on the
+	// number of redirects (10 at max is allowed by default).
```
Comment (roman-khimov) [inline pkg/services/oracle/network.go:90]: Isn't it `AllowAutoRedirect = false` which means no redirections allowed?

File: pkg/services/oracle/network.go [hunk 3]
```diff
@@ -86,10 +86,9 @@ func getDefaultClient(cfg config.OracleConfiguration) *http.Client {
...
 	client.CheckRedirect = func(req *http.Request, via []*http.Request) error {
-		if len(via) >= maxRedirections { // from https://github.com/neo-project/neo-modules/pull/694
-			return fmt.Errorf("%w: %d redirections are reached", ErrRestrictedRedirect, maxRedirections)
-		}
 		return nil
```
Comment (roman-khimov) [inline pkg/services/oracle/network.go:96]: So some error should be returned here?

File: pkg/services/oracle/network.go (hunk 2 above)
Comment (AnnaShaleva) [reply pkg/services/oracle/network.go:90]: It is, but at the same time redirections are handled by the C# node code itself and this code doesn't limit the number of redirections. So technically they are allowed.

Comment (roman-khimov) [reply pkg/services/oracle/network.go:90]: Wait, what?! It can't be unlimited!

File: pkg/services/oracle/network.go (hunk 3 above)
Comment (AnnaShaleva) [reply pkg/services/oracle/network.go:96]: But why? We do allow redirects, this code is made in order not to use the default redirect checker.

**Comment by AnnaShaleva** [comment]
Waiting for the decision on https://github.com/neo-project/neo-modules/pull/692#discussion_r822582682.

**Comment by roman-khimov** [comment]
Anything useful left here (http/https maybe)?

File: pkg/services/oracle/network.go (hunk 2 above)
Comment (AnnaShaleva) [reply pkg/services/oracle/network.go]: Resolved, limited by 2 including.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**